src/passes/references/arrayFunctions.ts

```diff
@@ -58,30 +63,49 @@ export class ArrayFunctions extends ReferenceSubPass {
 
     const baseType = getNodeType(node.vExpression, ast.compilerVersion);
     if (baseType instanceof PointerType && baseType.to instanceof ArrayType) {
-      if (baseType.location !== DataLocation.Storage && baseType.location !== DataLocation.Memory) {
-        throw new NotSupportedYetError(
-          `Accessing ${baseType.location} array length not implemented yet`,
-        );
-      }
+      if (baseType.location === DataLocation.CallData) {
+        const parent = node.parent;
+        const type = generalizeType(getNodeType(node, ast.compilerVersion))[0];
 
-      if (baseType.to.size !== undefined) {
-        const size = baseType.to.size.toString();
-        this.replace(
-          node,
-          createNumberLiteral(size, ast, 'uint256'),
-          undefined,
-          DataLocation.Default,
-          DataLocation.Default,
+        const funcStub = createCairoFunctionStub(
+          'felt_to_uint256',
+          [['cd_dstruct_array_len', typeNameFromTypeNode(type, ast)]],
+          [['len256', typeNameFromTypeNode(type, ast)]],
+          ['range_check_ptr'],
           ast,
+          node,
+          FunctionStateMutability.Pure,
+          FunctionStubKind.FunctionDefStub,
         );
-      } else {
-        const replacement =
-          baseType.location === DataLocation.Storage
-            ? ast.getUtilFuncGen(node).storage.dynArrayLength.gen(node, baseType.to)
-            : ast.getUtilFuncGen(node).memory.dynArrayLength.gen(node, ast);
-        // The length function returns the actual length rather than a storage pointer to it,
-        // so the new actual location is Default
-        this.replace(node, replacement, undefined, DataLocation.Default, expectedLoc, ast);
+
+        const funcCall = createCallToFunction(funcStub, [node], ast);
+
+        ast.registerImport(funcCall, 'warplib.maths.utils', 'felt_to_uint256');
+        ast.replaceNode(node, funcCall, parent);
```

Comment:
The reference subpasses use `this.replace` rather than `ast.replace` as it allows them to set the actual and expected location of the inserted node. For this, the actual should be default, and the expected should be the expected of the MemberAccess. Similarly the expected location of the member access should be set to Default